pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -775,6 +969,18 @@ public static PinotDataType getPinotDataTypeForIngestion(FieldSpec fieldSpec) {
         return fieldSpec.isSingleValueField() ? PinotDataType.FLOAT : PinotDataType.FLOAT_ARRAY;
       case DOUBLE:
         return fieldSpec.isSingleValueField() ? PinotDataType.DOUBLE : PinotDataType.DOUBLE_ARRAY;
+      case BOOLEAN:
+        if (fieldSpec.isSingleValueField()) {
+          return PinotDataType.BOOLEAN;
+        } else {
+          throw new IllegalStateException("There is no multi-value type for BOOLEAN");
```

Comment:
Any specific reason why no multi-value support for BOOLEAN? Boolean is a primitive type just like Numerical types?

Reply:
I feel it should be very rare to have MV boolean field (there are only 2 possible values, and the dimension should be either `true` or `false`). If required, we can add the support later. Added a todo here.